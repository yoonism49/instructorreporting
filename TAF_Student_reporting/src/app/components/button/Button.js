// @flow weak

import React      from 'react';
import PropTypes  from 'prop-types';
import cx         from 'classnames';

const Button = ({
  type,
  onClick,
  children
}) => {
  return (
    <button
      type="button"
      onClick={ onClick }
      className={
        cx({
          btn: true,
          'btn-default': type === 'default',
          'btn-primary': type === 'primary',
          'btn-success': type === 'success',
          'btn-info':    type === 'info',
          'btn-warning': type === 'warning',
          'btn-danger':  type === 'danger'
        })
      }>
      {  children }
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node,
  type:     PropTypes.oneOf(['default', 'primary', 'success', 'info', 'warning', 'danger']),
  onClick:  PropTypes.func
};

Button.defaultProps = {
  type: 'default'
};

export default Button;
