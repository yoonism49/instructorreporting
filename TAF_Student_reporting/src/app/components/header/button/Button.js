// @flow weak

import React      from 'react';
import PropTypes  from 'prop-types';

const Button = ({
  toggleSideMenu
}) => (
  <a
    href="#"
    onClick={toggleSideMenu}
    className="navbar-btn sidebar-toggle"
    data-toggle="offcanvas"
    role="button">
    <span className="sr-only">
      Toggle navigation
    </span>
    <span className="icon-bar" />
    <span className="icon-bar" />
    <span className="icon-bar" />
  </a>
);

Button.propTypes = {
  toggleSideMenu: PropTypes.func.isRequired
};

export default Button;
