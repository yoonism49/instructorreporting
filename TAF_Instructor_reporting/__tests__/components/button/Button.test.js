'use strict';

import React          from 'react';
import Button         from '../../../src/app/components/button/Button';
import renderer       from 'react-test-renderer';
import { mount }      from 'enzyme';

describe('Button component', () => {
  const onClickMock    = jest.fn();
  const preventDefault  = jest.fn();

  it('renders as expected', () => {
    const component = renderer.create(
      <div>
        <Button
          type="default"
        >
          <p>
            a button
          </p>
        </Button>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should call onClick on a click', () => {
    const mockClickEvent = { preventDefault };

    const component = mount(
      <Button
        type="default"
        onClick={onClickMock}
      />
    );
    component.simulate('click', mockClickEvent);
    expect(onClickMock).toHaveBeenCalled();
  });
});
