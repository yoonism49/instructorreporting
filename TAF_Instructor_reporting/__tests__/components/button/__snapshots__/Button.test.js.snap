// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders as expected 1`] = `
<div>
  <button
    className="btn btn-default"
    onClick={undefined}
    type="button"
  >
    <p>
      a button
    </p>
  </button>
</div>
`;
